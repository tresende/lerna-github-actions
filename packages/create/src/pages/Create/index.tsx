import React from 'react'

import * as S from './styles'

const Create = () => (
  <S.Wrapper>
    <h1>Create</h1>
  </S.Wrapper>
)

export default Create
