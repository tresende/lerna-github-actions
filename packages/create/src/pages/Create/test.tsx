import React from 'react'
import userEvent from '@testing-library/user-event'
import { render, screen } from '@testing-library/react'
import UserService from '@lerna-github-actions/core/src/services/Products'

import Create from '.'

jest.mock('@lerna-github-actions/core/src/services/Products')

describe('<Create />', () => {
  afterAll(() => {
    jest.resetAllMocks()
  })

  it('should render correctly', () => {
    const { container } = render(<Create />)

    screen.getByRole('heading', { name: /create/i })
    expect(container.firstChild).toMatchSnapshot()
  })

  it('Should call user service on subimit', () => {
    render(<Create />)
    const name = 'IPHONE 13'
    userEvent.type(screen.getByPlaceholderText(/Name/i), name)
    userEvent.click(screen.getByRole('button', { name: /save/i }))

    expect(UserService.save).toBeCalledTimes(1)
  })
})
