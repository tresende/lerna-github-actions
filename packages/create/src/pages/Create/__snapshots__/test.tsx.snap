// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Create /> should render correctly 1`] = `
<main
  class="sc-jRQBWg dEozX"
>
  <h1>
    Create
  </h1>
  <form>
    <div
      class="sc-bdvvtL iXMHdx"
    >
      <label
        class="sc-gsDKAQ iYacNX"
        for="Name"
      >
        Name
      </label>
      <input
        class="sc-dkPtRN iQffaK"
        name="Name"
        placeholder="Name"
        type="text"
        value=""
      />
    </div>
    <div
      class="sc-bdvvtL iXMHdx"
    >
      <label
        class="sc-gsDKAQ iYacNX"
        for="Description"
      >
        Description
      </label>
      <input
        class="sc-dkPtRN iQffaK"
        name="Description"
        placeholder="Description"
        type="text"
        value=""
      />
    </div>
    <div
      class="sc-bdvvtL iXMHdx"
    >
      <label
        class="sc-gsDKAQ iYacNX"
        for="Date"
      >
        Date
      </label>
      <input
        class="sc-dkPtRN iQffaK"
        name="Date"
        placeholder="Date"
        type="date"
        value=""
      />
    </div>
    <div
      class="sc-hKwDye cXwhOz"
    >
      <button
        class="sc-eCImPb cGGpCY"
        color="lightgray"
        type="button"
      >
        Clear
      </button>
      <button
        class="sc-eCImPb hKvVsk"
        color="#ea1d2c"
        type="submit"
      >
        Save
      </button>
    </div>
  </form>
</main>
`;
